### ISSUE TITLE
HTTP protocol not working

### ISSUE THREAD

**Issue by tukey42** [iobroker-community-adapters/ioBroker.philips-air#62]
**Describe the bug**  
I have an older purifier (AC3829/10) which most likely did not support COAP (don't know if there is a firmware update available).
Therefore I tested the HTTP protocol and found at least two bugs.
The first one happens directly when doing the Http Request, because the time is past as string and not as integer.
Line 74 of lib/http.js: options.aliveTimeout should most likely be this.aliveTimeout

After fixing that, most states are not updated because of incorrect mapping.
Line 74 of main.js: AirPurifier.getMapping is used in all cases (COAP and HTTP). For HTTP it should be AirHttpPurifier.getMapping. Even changing this in the source returns only very few states (e.g. pm25). Most others are still "(null)" (e.g. temp).

**To Reproduce**  
1. Just start the adapter in Http Mode

**Expected behavior**  
Connection is established and correct values are returned.

**Versions:**  
 - Adapter version: 1.0.1

**Comment by Apollon77** [comment]
Can you then please provide a debug log ... maybe we can then add missing mappings? We do not own such devices, so if you like a fix you need to provid eway more data

**Comment by tukey42** [comment]
Sure,
   here are the relevant lines from the log after fixing the two other mentioned issues, which are repeated every 30 seconds:

2022-12-22 12:49:29.478 - debug: philips-air.0 (2975518) STATUS: {"om":"0","pwr":"0","cl":false,"aqil":100,"uil":"0","dt":0,"dtrs":0,"mode":"A","func":"PH","rhset":50,"rh":41,"temp":26,"pm25":2,"iaql":1,"aqit":4,"ddp":"0","err":49408,"wl":0}
2022-12-22 12:49:29.654 - silly: philips-air.0 (2975518) States user redis pmessage philips-air.0.control.*/philips-air.0.control.mode:{"val":"A","ack":true,"ts":1671709769568,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671663984044}
2022-12-22 12:49:29.654 - debug: philips-air.0 (2975518) State change: {"val":"A","ack":true,"ts":1671709769568,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671663984044}
2022-12-22 12:49:59.479 - debug: philips-air.0 (2975518) Syncing...

And here is a screenshots of the corresponding states:



If you need any additional information or if you have a new version which I should test, just let me know.
Btw. switching power on/off by setting the control.power state works.

**Comment by Apollon77** [comment]
Please try GitHub version. I think I found the issue

**Comment by tukey42** [comment]
Great, many thanks! This looks much better.
In the log you just find some type mismatch messages.
Still some states are not set (e.g. the filter states, which are returned by this model). Here is a screenshot of the states:

**Comment by Apollon77** [comment]
Which log? Which screenshot? Email attachements are not added to the GitHub iossue - please add yourself!

**Comment by Apollon77** [comment]
(And please add a full debug log)

**Comment by tukey42** [comment]
Sorry, the screenshot was attached in the e-mail (which I understand came not through), but I forgot the log. No I try to add both through the web page.
<img width="719" alt="PastedGraphic-3" src="https://user-images.githubusercontent.com/26274365/209238663-36ab65fa-420b-455d-96bc-419efbac4b28.png">

Here is the log:
```
2022-12-22 23:17:09.177  - [32minfo[39m: host.iobroker instance system.adapter.philips-air.0 started with pid 4111390
2022-12-22 23:17:10.161  - [34mdebug[39m: philips-air.0 (4111390) Redis Objects: Use Redis connection: 127.0.0.1:9001
2022-12-22 23:17:10.240  - [34mdebug[39m: philips-air.0 (4111390) Objects client ready ... initialize now
2022-12-22 23:17:10.240  - [34mdebug[39m: philips-air.0 (4111390) Objects create System PubSub Client
2022-12-22 23:17:10.241  - [34mdebug[39m: philips-air.0 (4111390) Objects create User PubSub Client
2022-12-22 23:17:10.464  - [34mdebug[39m: philips-air.0 (4111390) Objects client initialize lua scripts
2022-12-22 23:17:10.472  - [34mdebug[39m: philips-air.0 (4111390) Objects connected to redis: 127.0.0.1:9001
2022-12-22 23:17:10.473  - [35msilly[39m: philips-air.0 (4111390) redis psubscribe cfg.o.system.user.*
2022-12-22 23:17:10.548  - [35msilly[39m: philips-air.0 (4111390) redis psubscribe cfg.o.enum.*
2022-12-22 23:17:10.550  - [35msilly[39m: philips-air.0 (4111390) objectDB connected
2022-12-22 23:17:10.551  - [34mdebug[39m: philips-air.0 (4111390) Redis States: Use Redis connection: 127.0.0.1:9000
2022-12-22 23:17:10.640  - [34mdebug[39m: philips-air.0 (4111390) States create System PubSub Client
2022-12-22 23:17:10.641  - [34mdebug[39m: philips-air.0 (4111390) States create User PubSub Client
2022-12-22 23:17:10.836  - [34mdebug[39m: philips-air.0 (4111390) States connected to redis: 127.0.0.1:9000
2022-12-22 23:17:10.836  - [35msilly[39m: philips-air.0 (4111390) statesDB connected
2022-12-22 23:17:11.240  - [34mdebug[39m: philips-air.0 (4111390) Plugin sentry Initialize Plugin (enabled=true)
2022-12-22 23:17:11.242  - [32minfo[39m: philips-air.0 (4111390) Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
2022-12-22 23:17:11.357  - [32minfo[39m: philips-air.0 (4111390) starting. Version 1.0.2 (non-npm: iobroker-community-adapters/ioBroker.philips-air) in /opt/iobroker/node_modules/iobroker.philips-air, node: v16.18.1, js-controller: 4.0.23
2022-12-22 23:17:11.402  - [35msilly[39m: philips-air.0 (4111390) States system redis pmessage system.adapter.philips-air.0.logLevel/system.adapter.philips-air.0.logLevel:{"val":"silly","ack":true,"ts":1671747431383,"q":0,"from":"system.adapter.philips-air.0","lc":1671236709622}
2022-12-22 23:17:11.419  - [34mdebug[39m: philips-air.0 (4111390) start with 10.2.1.6 {"host":"10.2.1.6","aliveTimeout":"30000","reconnectInterval":"30000","protocol":"http"}
2022-12-22 23:17:11.420  - [34mdebug[39m: philips-air.0 (4111390) started
2022-12-22 23:17:11.421  - [34mdebug[39m: philips-air.0 (4111390) Syncing...
2022-12-22 23:17:13.143  - [34mdebug[39m: philips-air.0 (4111390) Update client key: �^��n��}y�m<�M6�M}�m{��}��]��~
2022-12-22 23:17:13.145  - [34mdebug[39m: philips-air.0 (4111390) connected
2022-12-22 23:17:13.146  - [34mdebug[39m: philips-air.0 (4111390) STATUS: {"fanSpeed":"0","power":false,"childLock":"","lightBrightness":"100","buttonLight":false,"timerHours":"0","timerMinutes":"0","mode":"allergen","function":"humidification","targetHumidity":"50","humidity":"41","temperature":"26","pm25":"4","allergenIndex":"4","airQualityNotificationThreshold":"4","usedIndex":"iai","error":"no water","waterLevel":"0"}
2022-12-22 23:17:13.189  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.fanSpeed:{"val":"0","ack":true,"ts":1671747433188,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407628}
2022-12-22 23:17:13.189  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"0","ack":true,"ts":1671747433188,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407628}
2022-12-22 23:17:13.232  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.power:{"val":false,"ack":true,"ts":1671747433231,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407671}
2022-12-22 23:17:13.233  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":false,"ack":true,"ts":1671747433231,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407671}
2022-12-22 23:17:13.234  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.mode:{"val":"allergen","ack":true,"ts":1671747433233,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407719}
2022-12-22 23:17:13.234  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"allergen","ack":true,"ts":1671747433233,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407719}
2022-12-22 23:17:13.235  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.control.targetHumidity" has to be type "number" but received type "string" 
2022-12-22 23:17:13.236  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.targetHumidity:{"val":"50","ack":true,"ts":1671747433235,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407724}
2022-12-22 23:17:13.236  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"50","ack":true,"ts":1671747433235,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407724}
2022-12-22 23:17:13.280  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.function:{"val":true,"ack":true,"ts":1671747433279,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407767}
2022-12-22 23:17:13.281  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":true,"ack":true,"ts":1671747433279,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407767}
2022-12-22 23:17:13.281  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.control.lightBrightness" has to be type "number" but received type "string" 
2022-12-22 23:17:13.282  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.lightBrightness:{"val":"100","ack":true,"ts":1671747433281,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407770}
2022-12-22 23:17:13.282  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"100","ack":true,"ts":1671747433281,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407770}
2022-12-22 23:17:13.324  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.usedIndex:{"val":"iai","ack":true,"ts":1671747433323,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407816}
2022-12-22 23:17:13.325  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"iai","ack":true,"ts":1671747433323,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407816}
2022-12-22 23:17:13.327  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.buttonLight:{"val":false,"ack":true,"ts":1671747433326,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407859}
2022-12-22 23:17:13.332  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":false,"ack":true,"ts":1671747433326,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407859}
2022-12-22 23:17:13.332  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.control.timerHours" has to be type "number" but received type "string" 
2022-12-22 23:17:13.333  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.timerHours:{"val":"0","ack":true,"ts":1671747433333,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407862}
2022-12-22 23:17:13.333  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"0","ack":true,"ts":1671747433333,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407862}
2022-12-22 23:17:13.334  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.control.childLock" has to be type "boolean" but received type "string" 
2022-12-22 23:17:13.335  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.childLock:{"val":"","ack":true,"ts":1671747433334,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407908}
2022-12-22 23:17:13.335  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"","ack":true,"ts":1671747433334,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407908}
2022-12-22 23:17:13.335  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.control.airQualityNotificationThreshold" has to be type "number" but received type "string" 
2022-12-22 23:17:13.336  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.airQualityNotificationThreshold:{"val":"4","ack":true,"ts":1671747433336,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407963}
2022-12-22 23:17:13.336  - [34mdebug[39m: philips-air.0 (4111390) State change: {"val":"4","ack":true,"ts":1671747433336,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407963}
2022-12-22 23:17:13.337  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.humidity" has to be type "number" but received type "string" 
2022-12-22 23:17:13.380  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.allergenIndex" has to be type "number" but received type "string" 
2022-12-22 23:17:13.428  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.temperature" has to be type "number" but received type "string" 
2022-12-22 23:17:13.472  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.waterLevel" has to be type "number" but received type "string" 
2022-12-22 23:17:13.516  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.pm25" has to be type "number" but received type "string" 
2022-12-22 23:17:13.560  - [32minfo[39m: philips-air.0 (4111390) State value to set for "philips-air.0.status.timerMinutes" has to be type "number" but received type "string" 
2022-12-22 23:17:43.147  - [34mdebug[39m: philips-air.0 (4111390) Syncing...
2022-12-22 23:17:43.256  - [34mdebug[39m: philips-air.0 (4111390) STATUS: {"fanSpeed":"0","power":false,"childLock":"","lightBrightness":"100","buttonLight":false,"timerHours":"0","timerMinutes":"0","mode":"allergen","function":"humidification","targetHumidity":"50","humidity":"41","temperature":"26","pm25":"4","allergenIndex":"4","airQualityNotificationThreshold":"4","usedIndex":"iai","error":"no water","waterLevel":"0"}
2022-12-22 23:17:43.258  - [35msilly[39m: philips-air.0 (4111390) States user redis pmessage philips-air.0.control.*/philips-air.0.control.fanSpeed:{"val":"0","ack":true,"ts":1671747463257,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407628}

```

**Comment by Apollon77** [comment]
Ok, I weill check the type stuff, but the filter infos are not contained in the data returned from the device, so yes the states (that are always created) stay null

**Comment by tukey42** [comment]
Beside of the getStatus() call, there is a getFilters() call, which would return the filter status. This is of course more of a feature request....

**Comment by Apollon77** [comment]
Ahh this might be a difference between Coap and HTTP ... will check

**Comment by Apollon77** [comment]
@tukey42 Try GitHub version again please and post debug log in any case!
HTTP is now requesting Firmware, Wifi and Filters also

**Comment by tukey42** [comment]
Wow, great, it seems, that at least all states I like to see are now set. Many thanks!

Here is again a screenshot and the log:
```
2022-12-23 13:18:42.787  - [32minfo[39m: host.iobroker instance system.adapter.philips-air.0 started with pid 626762
2022-12-23 13:18:44.055  - [34mdebug[39m: philips-air.0 (626762) Redis Objects: Use Redis connection: 127.0.0.1:9001
2022-12-23 13:18:44.147  - [34mdebug[39m: philips-air.0 (626762) Objects client ready ... initialize now
2022-12-23 13:18:44.147  - [34mdebug[39m: philips-air.0 (626762) Objects create System PubSub Client
2022-12-23 13:18:44.148  - [34mdebug[39m: philips-air.0 (626762) Objects create User PubSub Client
2022-12-23 13:18:44.360  - [34mdebug[39m: philips-air.0 (626762) Objects client initialize lua scripts
2022-12-23 13:18:44.372  - [34mdebug[39m: philips-air.0 (626762) Objects connected to redis: 127.0.0.1:9001
2022-12-23 13:18:44.373  - [35msilly[39m: philips-air.0 (626762) redis psubscribe cfg.o.system.user.*
2022-12-23 13:18:44.439  - [35msilly[39m: philips-air.0 (626762) redis psubscribe cfg.o.enum.*
2022-12-23 13:18:44.441  - [35msilly[39m: philips-air.0 (626762) objectDB connected
2022-12-23 13:18:44.443  - [34mdebug[39m: philips-air.0 (626762) Redis States: Use Redis connection: 127.0.0.1:9000
2022-12-23 13:18:44.542  - [34mdebug[39m: philips-air.0 (626762) States create System PubSub Client
2022-12-23 13:18:44.542  - [34mdebug[39m: philips-air.0 (626762) States create User PubSub Client
2022-12-23 13:18:44.728  - [34mdebug[39m: philips-air.0 (626762) States connected to redis: 127.0.0.1:9000
2022-12-23 13:18:44.728  - [35msilly[39m: philips-air.0 (626762) statesDB connected
2022-12-23 13:18:45.106  - [34mdebug[39m: philips-air.0 (626762) Plugin sentry Initialize Plugin (enabled=true)
2022-12-23 13:18:45.108  - [32minfo[39m: philips-air.0 (626762) Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
2022-12-23 13:18:45.238  - [32minfo[39m: philips-air.0 (626762) starting. Version 1.0.2 (non-npm: iobroker-community-adapters/ioBroker.philips-air) in /opt/iobroker/node_modules/iobroker.philips-air, node: v16.18.1, js-controller: 4.0.23
2022-12-23 13:18:45.255  - [35msilly[39m: philips-air.0 (626762) States system redis pmessage system.adapter.philips-air.0.logLevel/system.adapter.philips-air.0.logLevel:{"val":"silly","ack":true,"ts":1671797925252,"q":0,"from":"system.adapter.philips-air.0","lc":1671236709622}
2022-12-23 13:18:45.264  - [34mdebug[39m: philips-air.0 (626762) start with 10.2.1.6 {"host":"10.2.1.6","aliveTimeout":"30000","reconnectInterval":"30000","protocol":"http"}
2022-12-23 13:18:45.264  - [34mdebug[39m: philips-air.0 (626762) started
2022-12-23 13:18:45.267  - [34mdebug[39m: philips-air.0 (626762) Syncing...
2022-12-23 13:18:47.001  - [34mdebug[39m: philips-air.0 (626762) Update client key: �^��n��}y�m<�M6�M}�m{��}��]��~
2022-12-23 13:18:47.042  - [34mdebug[39m: philips-air.0 (626762) connected
2022-12-23 13:18:47.042  - [34mdebug[39m: philips-air.0 (626762) STATUS: {"fanSpeed":"0","power":false,"childLock":"","lightBrightness":"100","buttonLight":false,"timerHours":"0","timerMinutes":"0","mode":"allergen","function":"humidification","targetHumidity":"50","humidity":"42","temperature":"26","pm25":"9","allergenIndex":"3","airQualityNotificationThreshold":"4","usedIndex":"iai","error":"no water","waterLevel":"0"}
2022-12-23 13:18:47.063  - [34mdebug[39m: philips-air.0 (626762) STATUS: {"version":"15","upgrade":"","state":"idle","progress":0,"statusmsg":"","mandatory":false,"name":"AC3829_10"}
2022-12-23 13:18:47.066  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.fanSpeed:{"val":"0","ack":true,"ts":1671797927065,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407628}
2022-12-23 13:18:47.067  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"0","ack":true,"ts":1671797927065,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407628}
2022-12-23 13:18:47.096  - [34mdebug[39m: philips-air.0 (626762) STATUS: {"hepaFilterType":"NanoProtect Filter Series 3 (FY2422)","activeCarbonFilterType":"NanoProtect Filter AC (FY2420)","preFilterCleanInHours":"69","hepaFilterReplaceInHours":"4147","activeCarbonFilterReplaceInHours":"4147","wickFilterReplaceInHours":"4147"}
2022-12-23 13:18:47.123  - [34mdebug[39m: philips-air.0 (626762) STATUS: {"ssid":"IoT","password":"","protection":"wpa-2","ipaddress":"10.2.1.6","netmask":"255.255.0.0","gateway":"10.2.0.1","dhcp":true,"macaddress":"e8:c1:d7:01:54:03","cppid":"e8c1d7fffe015403"}
2022-12-23 13:18:47.127  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.power:{"val":false,"ack":true,"ts":1671797927124,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407671}
2022-12-23 13:18:47.127  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":false,"ack":true,"ts":1671797927124,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407671}
2022-12-23 13:18:47.142  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.mode:{"val":"allergen","ack":true,"ts":1671797927141,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407719}
2022-12-23 13:18:47.142  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"allergen","ack":true,"ts":1671797927141,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407719}
2022-12-23 13:18:47.187  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.control.targetHumidity" has to be type "number" but received type "string" 
2022-12-23 13:18:47.190  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.targetHumidity:{"val":"50","ack":true,"ts":1671797927188,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407724}
2022-12-23 13:18:47.190  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"50","ack":true,"ts":1671797927188,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407724}
2022-12-23 13:18:47.191  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.filter.preFilterCleanInHours" has to be type "number" but received type "string" 
2022-12-23 13:18:47.193  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.function:{"val":true,"ack":true,"ts":1671797927193,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407767}
2022-12-23 13:18:47.194  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":true,"ack":true,"ts":1671797927193,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407767}
2022-12-23 13:18:47.194  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.filter.hepaFilterReplaceInHours" has to be type "number" but received type "string" 
2022-12-23 13:18:47.195  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.control.lightBrightness" has to be type "number" but received type "string" 
2022-12-23 13:18:47.197  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.lightBrightness:{"val":"100","ack":true,"ts":1671797927196,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407770}
2022-12-23 13:18:47.198  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"100","ack":true,"ts":1671797927196,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407770}
2022-12-23 13:18:47.198  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.filter.activeCarbonFilterReplaceInHours" has to be type "number" but received type "string" 
2022-12-23 13:18:47.201  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.usedIndex:{"val":"iai","ack":true,"ts":1671797927200,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407816}
2022-12-23 13:18:47.201  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"iai","ack":true,"ts":1671797927200,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407816}
2022-12-23 13:18:47.201  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.filter.wickFilterReplaceInHours" has to be type "number" but received type "string" 
2022-12-23 13:18:47.203  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.buttonLight:{"val":false,"ack":true,"ts":1671797927202,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407859}
2022-12-23 13:18:47.205  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":false,"ack":true,"ts":1671797927202,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407859}
2022-12-23 13:18:47.252  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.control.timerHours" has to be type "number" but received type "string" 
2022-12-23 13:18:47.253  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.timerHours:{"val":"0","ack":true,"ts":1671797927252,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407862}
2022-12-23 13:18:47.253  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"0","ack":true,"ts":1671797927252,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407862}
2022-12-23 13:18:47.296  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.control.childLock" has to be type "boolean" but received type "string" 
2022-12-23 13:18:47.297  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.childLock:{"val":"","ack":true,"ts":1671797927296,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407908}
2022-12-23 13:18:47.297  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"","ack":true,"ts":1671797927296,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407908}
2022-12-23 13:18:47.344  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.control.airQualityNotificationThreshold" has to be type "number" but received type "string" 
2022-12-23 13:18:47.345  - [35msilly[39m: philips-air.0 (626762) States user redis pmessage philips-air.0.control.*/philips-air.0.control.airQualityNotificationThreshold:{"val":"4","ack":true,"ts":1671797927344,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407963}
2022-12-23 13:18:47.345  - [34mdebug[39m: philips-air.0 (626762) State change: {"val":"4","ack":true,"ts":1671797927344,"q":0,"from":"system.adapter.philips-air.0","user":"system.user.admin","lc":1671747407963}
2022-12-23 13:18:47.392  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.humidity" has to be type "number" but received type "string" 
2022-12-23 13:18:47.436  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.allergenIndex" has to be type "number" but received type "string" 
2022-12-23 13:18:47.480  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.temperature" has to be type "number" but received type "string" 
2022-12-23 13:18:47.528  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.waterLevel" has to be type "number" but received type "string" 
2022-12-23 13:18:47.576  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.pm25" has to be type "number" but received type "string" 
2022-12-23 13:18:47.624  - [32minfo[39m: philips-air.0 (626762) State value to set for "philips-air.0.status.timerMinutes" has to be type "number" but received type "string" 
2022-12-23 13:19:17.131  - [34mdebug[39m: philips-air.0 (626762) Syncing...
```

<img width="1448" alt="Bildschirmfoto 2022-12-23 um 13 25 02" src="https://user-images.githubusercontent.com/26274365/209336557-e3d30709-00cb-4daa-b437-a329db4ab87b.png">
<img width="1448" alt="Bildschirmfoto 2022-12-23 um 13 25 28" src="https://user-images.githubusercontent.com/26274365/209336574-195c90bb-f17c-4b77-aa0c-dd40a21cd6d5.png">

**Comment by Apollon77** [comment]
Released as 1.0.3
